CompositionRoot/A.swift

```diff
@@ -4,15 +4,20 @@ protocol AProtocol {
 
 class A: AProtocol {
     
-    let applicationContext: ApplicationContext
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    let f: FProtocol
+    let g: GProtocol
+    let bFactory: BFactoryProtocol
+
+    init(f: FProtocol, g: GProtocol, bFactory: BFactoryProtocol) {
+        self.f = f
+        self.g = g
+        self.bFactory = bFactory
     }
     
     func aMethod() {
-        let b = B(applicationContext: applicationContext)
```

Comment:
This line of code makes A depend on the specific class `B`. Ideally, It should depends on abstract protocol.
One of the SOLID principles, Dependency Inversion principle, requires both high level class and low level depend on abstract interface, not specific class. So we should depend on abstract protocol